addons/mrp/models/mrp_production.py

```diff
@@ -850,18 +856,13 @@ def _can_produce_serial_number(self, sn=None):
         sn = sn or self.lot_producing_id
         if self.product_id.tracking == 'serial' and sn:
             message, dummy = self.env['stock.quant'].sudo()._check_serial_number(self.product_id, sn, self.company_id)
+            is_reproducing = self.lot_producing_id and self.lot_producing_id.id in self.move_raw_ids.lot_ids.ids
+            if is_reproducing:
```

Comment:
You can move that condition on the above `if`. Not even useful to search the quants if we're going to ignore the results anyway